File: src/js/main/RichJson_keepKeyCommands.js
import {cloneObject} from "./RichJsonHelper";
import {__RICH_JSON_KEY_COMMAND_MEMBER} from "./RichJson";

/**
 * Make key commands constant for given JSON object.
 * @param jsonObject
 * @returns {{}}
 */
export function keepKeyCommands(jsonObject) {
    if (Object.hasOwn(jsonObject, __RICH_JSON_KEY_COMMAND_MEMBER)) {
        jsonObject[__RICH_JSON_KEY_COMMAND_MEMBER] = cloneObject(jsonObject[__RICH_JSON_KEY_COMMAND_MEMBER]);
    }
    return jsonObject;
}

File: src/python/main/helper/rich_json_keep_key_commands.py
from .rich_json_helper import clone_object
from ..core.rich_json import _RICH_JSON_KEY_COMMAND_MEMBER


def keep_key_commands(json_object):
    """
    Make key commands constant for the given JSON object by creating a deep copy
    of the key command member.
    """
    if isinstance(json_object, dict) and _RICH_JSON_KEY_COMMAND_MEMBER in json_object:
        json_object[_RICH_JSON_KEY_COMMAND_MEMBER] = clone_object(json_object[_RICH_JSON_KEY_COMMAND_MEMBER])

    return json_object
